Packer crashes with specified subnet_id

## Packer Version

Packer v1.7.0

## OS

Ubuntu Server 20.04 LTS

## Issue Overview

When only `subnet_id` is specified, packer crashes because created new VPC does not match the specified subnet.

## Expected Behavior

Packer does not create a new VPC, just use the specified subnet to launch KEC instance.

## Example Code

```hcl
packer {
  required_plugins {
    ksyun = {
      version = ">= 0.0.2"
      source  = "github.com/kingsoftcloud/ksyun"
    }
  }
}

locals {
  # format: 20210126T024552Z
  timestamp = regex_replace(timestamp(), "[- :]", "")
}

source "ksyun-kec" "focal-amd64" {
  region = "cn-beijing-6"

  source_image_id = "IMG-6d858fc8-45a5-44fd-9053-dbbf6def45da"
  image_type      = "LocalImage"
  image_name      = "ubuntu-2004-x86_64-${local.timestamp}"

  subnet_id                 = "5c3cca76-b248-416e-994f-3f6cb168ee58"
  instance_type             = "S4.1A"
  instance_charge_type      = "HourlyInstantSettlement"
  pause_before_connecting   = "30s"
  communicator              = "ssh"
  ssh_username              = "ubuntu"
  ssh_clear_authorized_keys = true

  system_disk_type = "SSD3.0"
  system_disk_size = 20

  associate_public_ip_address = true
  public_ip_charge_type       = "HourlyInstantSettlement"
  public_ip_band_width        = 1
}
```

## Error Output

```
$ packer build -timestamp-ui templates/ubuntu/
ksyun-kec.focal-amd64: output will be in this color.

2021-03-22T11:56:20+08:00:     ksyun-kec.focal-amd64: Found image ID: IMG-6d858fc8-45a5-44fd-9053-dbbf6def45da
2021-03-22T11:56:20+08:00: ==> ksyun-kec.focal-amd64: Using SSH Agent for create new key pair packer_60581564-8b4e-087e-8b9c-e96908e9e010
2021-03-22T11:56:20+08:00: ==> ksyun-kec.focal-amd64: Creating new Vpc with name ksyun_packer_vpc cidr 172.20.0.0/16
2021-03-22T11:56:22+08:00: ==> ksyun-kec.focal-amd64: Subnet id 5c3cca76-b248-416e-994f-3f6cb168ee58 vpc not match
2021-03-22T11:56:22+08:00: ==> ksyun-kec.focal-amd64: Deleting Vpc with Id 3abde4d4-9eca-4e82-8add-05dd79056cd0
2021-03-22T11:56:23+08:00: Build 'ksyun-kec.focal-amd64' errored after 2 seconds 897 milliseconds: unexpected EOF

==> Wait completed after 2 seconds 897 milliseconds

==> Some builds didn't complete successfully and had errors:
--> ksyun-kec.focal-amd64: unexpected EOF

==> Builds finished but no artifacts were created.
panic: interface conversion: interface {} is nil, not string
2021/03/22 11:56:23 packer-plugin-ksyun_v0.0.2_x5.0_linux_amd64 plugin:
2021/03/22 11:56:23 packer-plugin-ksyun_v0.0.2_x5.0_linux_amd64 plugin: goroutine 73 [running]:
2021/03/22 11:56:23 packer-plugin-ksyun_v0.0.2_x5.0_linux_amd64 plugin: github.com/kingsoftcloud/packer-plugin-ksyun/builder/kec.(*Builder).Run(0xc0004e1080, 0x144d3c0, 0xc000010300, 0x1453820, 0xc0004b8180, 0x1432c40, 0xc0004ee260, 0x0, 0xc0003003f8, 0xc000104bd0, ...)
2021/03/22 11:56:23 packer-plugin-ksyun_v0.0.2_x5.0_linux_amd64 plugin:         github.com/kingsoftcloud/packer-plugin-ksyun/builder/kec/builder.go:125 +0xb85
2021/03/22 11:56:23 packer-plugin-ksyun_v0.0.2_x5.0_linux_amd64 plugin: github.com/hashicorp/packer-plugin-sdk/rpc.(*BuilderServer).Run(0xc00030c140, 0x1, 0xc0003100a0, 0x0, 0x0)
2021/03/22 11:56:23 packer-plugin-ksyun_v0.0.2_x5.0_linux_amd64 plugin:         github.com/hashicorp/packer-plugin-sdk@v0.0.14/rpc/builder.go:117 +0x1c4
2021/03/22 11:56:23 packer-plugin-ksyun_v0.0.2_x5.0_linux_amd64 plugin: reflect.Value.call(0xc0003003c0, 0xc000308098, 0x13, 0x1258db2, 0x4, 0xc000104f08, 0x3, 0x3, 0x0, 0x0, ...)
2021/03/22 11:56:23 packer-plugin-ksyun_v0.0.2_x5.0_linux_amd64 plugin:         reflect/value.go:476 +0x8c7
2021/03/22 11:56:23 packer-plugin-ksyun_v0.0.2_x5.0_linux_amd64 plugin: reflect.Value.Call(0xc0003003c0, 0xc000308098, 0x13, 0xc0004a8f08, 0x3, 0x3, 0x0, 0x0, 0x0)
2021/03/22 11:56:23 packer-plugin-ksyun_v0.0.2_x5.0_linux_amd64 plugin:         reflect/value.go:337 +0xb9
2021/03/22 11:56:23 packer-plugin-ksyun_v0.0.2_x5.0_linux_amd64 plugin: net/rpc.(*service).call(0xc00030c180, 0xc00032e050, 0xc0000354c0, 0xc0000354d0, 0xc00032a200, 0xc00000df20, 0x1058340, 0xc00031008c, 0x18a, 0x10362e0, ...)
2021/03/22 11:56:23 packer-plugin-ksyun_v0.0.2_x5.0_linux_amd64 plugin:         net/rpc/server.go:377 +0x189
2021/03/22 11:56:23 packer-plugin-ksyun_v0.0.2_x5.0_linux_amd64 plugin: created by net/rpc.(*Server).ServeCodec
2021/03/22 11:56:23 packer-plugin-ksyun_v0.0.2_x5.0_linux_amd64 plugin:         net/rpc/server.go:474 +0x445
2021/03/22 11:56:23 /home/wangzhiguang/.packer.d/plugins/github.com/kingsoftcloud/ksyun/packer-plugin-ksyun_v0.0.2_x5.0_linux_amd64: plugin process exited
2021/03/22 11:56:23 ui error: 2021-03-22T11:56:23+08:00: Build 'ksyun-kec.focal-amd64' errored after 2 seconds 897 milliseconds: unexpected EOF
2021/03/22 11:56:23 ui:
==> Wait completed after 2 seconds 897 milliseconds
2021/03/22 11:56:23 machine readable: error-count []string{"1"}
2021/03/22 11:56:23 ui error:
==> Some builds didn't complete successfully and had errors:
2021/03/22 11:56:23 machine readable: ksyun-kec.focal-amd64,error []string{"unexpected EOF"}
2021/03/22 11:56:23 ui error: --> ksyun-kec.focal-amd64: unexpected EOF
2021/03/22 11:56:23 ui:
==> Builds finished but no artifacts were created.
2021/03/22 11:56:23 waiting for all plugin processes to complete...
2021/03/22 11:56:23 /usr/local/bin/packer: plugin process exited
2021/03/22 11:56:23 /usr/local/bin/packer: plugin process exited
2021/03/22 11:56:23 /usr/local/bin/packer: plugin process exited
2021/03/22 11:56:23 /usr/local/bin/packer: plugin process exited
2021/03/22 11:56:23 /usr/local/bin/packer: plugin process exited
2021/03/22 11:56:23 /usr/local/bin/packer: plugin process exited
2021/03/22 11:56:23 /usr/local/bin/packer: plugin process exited
2021/03/22 11:56:23 /usr/local/bin/packer: plugin process exited
2021/03/22 11:56:23 /usr/local/bin/packer: plugin process exited
2021/03/22 11:56:23 /usr/local/bin/packer: plugin process exited
2021/03/22 11:56:23 /usr/local/bin/packer: plugin process exited
2021/03/22 11:56:23 /usr/local/bin/packer: plugin process exited



!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Packer crashed! This is always indicative of a bug within Packer.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Packer[1] so that we can fix this.

[1]: https://github.com/hashicorp/packer/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
$
```

--- a/README.md
+++ b/README.md
@@ -30,6 +30,18 @@ Link the build to Packer
 ```sh
 $ln -s $GOPATH/bin/packer-plugin-ksyun ~/.packer.d/plugins/packer-plugin-ksyun
 ```
+### Install from HCL:
+```hcl
+packer {
+  required_plugins {
+    ksyun = {
+      version = ">= 0.0.2"
+      source  = "github.com/kingsoftcloud/ksyun"
+    }
+  }
+}
+```
+
 
 ### Install from release:
 

--- a/builder/kec/builder.go
+++ b/builder/kec/builder.go
@@ -6,6 +6,7 @@ package kec
 
 import (
 	"context"
+	"fmt"
 	"github.com/hashicorp/hcl/v2/hcldec"
 	"github.com/hashicorp/packer-plugin-sdk/common"
 	"github.com/hashicorp/packer-plugin-sdk/communicator"
@@ -119,6 +120,12 @@ func (b *Builder) Run(ctx context.Context, ui packersdk.Ui, hook packersdk.Hook)
 	// Run!
 	b.runner = commonsteps.NewRunner(steps, b.config.PackerConfig, ui)
 	b.runner.Run(ctx, stateBag)
+
+	// If there was an error, return that
+	if err, ok := stateBag.GetOk("error"); ok {
+		ui.Say(fmt.Sprintf("find some error %v ", err))
+		return nil, err.(error)
+	}
 
 	// Build the artifact and return it
 	artifact := &Artifact{

--- a/builder/kec/step_create_image.go
+++ b/builder/kec/step_create_image.go
@@ -19,6 +19,9 @@ func (s *stepCreateKsyunImage) Run(ctx context.Context, stateBag multistep.State
 	createImage := make(map[string]interface{})
 	createImage["InstanceId"] = instanceId
 	createImage["Name"] = s.KsyunImageConfig.KsyunImageName
+	if s.KsyunImageConfig.KsyunImageType != "" {
+		createImage["Type"] = s.KsyunImageConfig.KsyunImageType
+	}
 	resp, errorCreate := client.KecClient.CreateImage(&createImage)
 	if errorCreate != nil {
 		return Halt(stateBag, errorCreate, "Error creating  kec image")

--- a/builder/kec/utils.go
+++ b/builder/kec/utils.go
@@ -44,7 +44,7 @@ func getSdkValue(stateBag multistep.StateBag, keyPattern string, obj interface{}
 			if err != nil {
 				Halt(stateBag, err, fmt.Sprintf("keyPattern %s index %d must number", keyPattern, index))
 			}
-			if len(root.([]interface{})) < i {
+			if len(root.([]interface{})) <= i {
 				return nil
 			}
 			root = root.([]interface{})[i]

